lpc55_sign_bin/src/main.rs

```diff
@@ -381,7 +414,7 @@ fn main() -> Result<()> {
             certs,
         } => {
             let cfg = certs.try_into_config()?;
-            let private_key = read_rsa_private_key(&cfg.private_key)?;
+            let private_key = read_rsa_private_key(cfg.private_key.as_ref().unwrap())?;
```

Comment:
This `unwrap` fires if called with no arguments:
```
cargo run --bin=lpc55_sign -- sign-image --in ~/Desktop/final_old.bin --out out.bin
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/lpc55_sign sign-image --in /Users/mjk/Desktop/final_old.bin --out /Users/mjk/Desktop/out.bin`
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', lpc55_sign_bin/src/main.rs:417:77
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Hm, that's probably annoying. Lemme see what I can do about that.

Reply:
Ah, yeah, so, this is one of the unfortunate things about reusing a chunk of flags with `clap(flatten)` when some of the flags are mandatory in some circumstances but not others -- we don't get any polite help in this.

I can get this behavior, if that seems ok:

```
Error: sign-image requires a private key to be provided as an arg or in the cert-cfg.
```

Reply:
Yeah, that seems totally fine.

(I've long-since resigned myself to working around `clap` corner cases in user code)